source/Halibut/ServiceModel/HalibutProxyWithAsync.cs

```diff
@@ -72,9 +69,7 @@ async Task<(MethodInfo, object)> MakeRpcCall(MethodInfo asyncMethod, object[] ar
 
             var request = CreateRequest(asyncMethod, serviceMethod, args);
 
-            using var requestCancellationTokens = RequestCancellationTokens(halibutProxyRequestOptions);
-
-            var response = await messageRouter(request, serviceMethod, requestCancellationTokens);
+            var response = await messageRouter(request, serviceMethod, halibutProxyRequestOptions?.RequestCancellationToken ?? CancellationToken.None);
```

Comment:
Just wondering, what situation would `halibutProxyRequestOptions` be null in? I thought put this on all async methods?

Reply:
(I'm seeing if we can simplify it and ensure `halibutProxyRequestOptions` will always be there 😁 )

Reply:
Good question, short answer I'm not sure without more investigation, I assumed it would never be null originally, and it blew up a lot of tests so reverted to the original logic of defaulting if null.

Reply:
A quick look shows that we do not enforce that a HalibutProxyRequestOptions has to be provided when making an RPC call and 100's of our tests don't provide it.

It probably makes sense to charge this and enforce the parameter is supplied so it's used in the best way e.g. cancellation tokens passed etc. I'll card it.